pkg/xsysinfo/nvidia_gpu.go

```diff
@@ -0,0 +1,75 @@
+//go:build !arm && !arm64
+// +build !arm,!arm64
+
+package xsysinfo
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+type GPUInfo struct {
+	DeviceName        string  `json:"device_name"`
+	TotalMemory       uint64  `json:"total_memory"`
+	FreeMemory        uint64  `json:"free_memory"`
+	UsedMemory        uint64  `json:"used_memory"`
+	MemoryUtilization float32 `json:"memory_utilization"`
+}
+
+// GetNvidiaGpuInfo uses pkg nvml is a go binding around C API provided by libnvidia-ml.so
+// to fetch GPU stats
+func GetNvidiaGpuInfo() ([]GPUInfo, error) {
```

Comment:
Minor nit, but it's good practice to keep acronyms uppercase, e.g. `UnmarshalYAML`, `GetXYZ`:

```suggestion
func GetNvidiaGPUInfo() ([]GPUInfo, error) {
```